staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1186,6 +1190,36 @@ func schemaIsNonStructural(schema *apiextensions.JSONSchemaProps) bool {
 	return len(structuralschema.ValidateStructural(nil, ss)) > 0
 }
 
+// requirePrunedDefaults returns true if the old CRD spec had no unpruned defaults.
+func requirePrunedDefaults(spec *apiextensions.CustomResourceDefinitionSpec) bool {
```

Comment:
Can we add an initial check that the old schema has defaults at all (specHasDefaults), and if it doesn't, return true? That makes it a lot simpler to verify we're not going to let in unpruned defaults to a CRD that currently doesn't even have defaults